src/Eryph.GenePool.Model/ApiKeyName.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Dbosoft.Functional.DataTypes;
+using Eryph.ConfigModel;
+using LanguageExt.ClassInstances;
+
+namespace Eryph.GenePool.Model;
+
+public class ApiKeyName : ValidatingNewType<ApiKeyName, string, OrdStringOrdinalIgnoreCase>
+{
+    public ApiKeyName(string value) : base(value)
+    {
+        ValidOrThrow(Validations<ApiKeyName>.ValidateCharacters(
```

Comment:
@fw2568 I would like to add some kind of validation for the API key name as it will be used for the CLI validation. How restrictive should we be here?